documentcloud/addons/models.py

```diff
@@ -577,6 +581,46 @@ def dispatch(self, document_pk=None):
             )
 
 
+class AddOnDisableLog(models.Model):
+    """Log usage of Add-On Disables"""
+
+    # Reference to the AddOnEvent
+    addon_event = models.ForeignKey(
+        verbose_name=_("event reference"),
+        to="addons.AddOnEvent",
+        on_delete=models.PROTECT,
+        related_name="disable_logs",
+        help_text=_("Reference to the Add-On Event"),
+    )
+
+    created_at = AutoCreatedField(
+        _("created at"),
+        help_text=_("Timestamp of when the add-on disable log was created"),
+        db_index=True,
+    )
+    previous_event_state = models.IntegerField(
+        _("previous event state"),
+        choices=Event.choices,
+        help_text=_("The event state of the Add-On before disable"),
+    )
+
+    reverted = models.BooleanField(
+        _("reverted"),
+        default=False,
+        help_text=_("Indicates whether this disable log has been reverted"),
+    )
+
+    def revert_event(self):
```

Comment:
Add `@transaction.atomic` decorator to this method
It makes the whole method into a database transaction - this means that if for some reason it crashes, either both database saves go though, or neither of them do.  This prevents the case of reverting the event without logging it successfully.